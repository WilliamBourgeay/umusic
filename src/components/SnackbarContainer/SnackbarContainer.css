.success {
  background-color: darkgreen !important;
}

.error {
  background-color: darkred !important;
}

.SnackbarContainer {
  justify-content: center;
}