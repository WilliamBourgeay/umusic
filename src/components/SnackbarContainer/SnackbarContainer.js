// @flow

import React from 'react';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import Snackbar from '@material-ui/core/Snackbar';
import './SnackbarContainer.css';

type Props = {
  message: string,
  variant: string,
};

type State = {
  open: boolean,
};

class SnackbarContainer extends React.Component<Props, State> {
  state = {
    open: false,
  };

  componentDidUpdate(prevProps: Props) {
    const { message } = this.props;
    const { open } = this.state;

    if (!open && prevProps.message !== message && message !== '')
      this.setState({ open: true }); // eslint-disable-line
  }

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { message, variant } = this.props;
    const { open } = this.state;

    return (
      <Snackbar
        autoHideDuration={5000}
        anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
        open={open}
        onClose={this.handleClose}
      >
        <SnackbarContent
          className={`SnackbarContainer ${variant}`}
          message={<span className="textSnackbarContainer">{message}</span>}
        />
      </Snackbar>
    );
  }
}

export default SnackbarContainer;
