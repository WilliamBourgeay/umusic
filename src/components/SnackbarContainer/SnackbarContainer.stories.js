import React from 'react';
import { storiesOf } from '@storybook/react';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContainer from './SnackbarContainer';

const stories = storiesOf('Snackbar', module);

stories
  .add('Snackbar Error', () => (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open="true"
    >
      <SnackbarContainer
        message="Error message"
        variant="error"
      />
    </Snackbar>
  ))
  .add('Snackbar Success', () => (
    <Snackbar
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      open="true"
    >
      <SnackbarContainer
        message="Success message"
        variant="success"
      />
    </Snackbar>
  ));
