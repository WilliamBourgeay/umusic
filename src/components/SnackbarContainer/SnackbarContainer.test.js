import * as React from 'react';
import renderer from 'react-test-renderer';

import SnackbarContainer from './SnackbarContainer';

describe('SnackbarContainer', () => {
  it('should render a SnackbarContainer error', () => {
    const tree = renderer.create(<SnackbarContainer variant="error" message="Error" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should render a SnackbarContainer success', () => {
    const tree = renderer.create(<SnackbarContainer variant="error" message="Error" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
