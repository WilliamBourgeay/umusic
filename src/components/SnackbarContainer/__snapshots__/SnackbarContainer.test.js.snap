// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarContainer should render a SnackbarContainer error 1`] = `null`;

exports[`SnackbarContainer should render a SnackbarContainer success 1`] = `null`;
